force-app/main/default/classes/RD2_ETableController.cls

```diff
@@ -129,22 +131,38 @@ public with sharing class RD2_ETableController {
     * @return List<RD2_ETableController.TableView> returns processed Recurring Donations.
     */
     private List<RD2_ETableController.TableView> getTableViews() {
-        List<RD2_ETableController.TableView> recurringDonations = new List<RD2_ETableController.TableView>();
+        Map<Id, RD2_ETableController.TableView> recurringDonations = new Map<Id,RD2_ETableController.TableView>();
         if(RD2_EnablementService.isRecurringDonations2Enabled) {
             if(controller.hasAccessToRecurringDonation()) {
+                Set<Id> fixedRecurringDonationIds = new Set<Id>();
+
                 for(npe03__Recurring_Donation__c recDonation : selector.getRecurringDonations()) {
-                    recurringDonations.add( new RD2_ETableController.TableView( recDonation,
+                    recurringDonations.put( recDonation.Id, new RD2_ETableController.TableView( recDonation,
```

Comment:
Can we exclude the `finalDonation` parameter here since we're just setting a `null` value? We can also remove it from the `TableView` constructor. If we expect to add more optional parameters to the `TableView` constructor, then we should consider building the TableView object using the builder pattern. The `TEST_RecurringDonationBuilder` class is a good example.

Reply:
Thanks Victor, I updated the inner class constructor.